app/lib/common/widgets/html_editor/components/mention_list.dart

```diff
@@ -0,0 +1,281 @@
+  });
+
+  final EditorState editorState;
+  final String roomId;
+  final MentionType mentionType;
+  final VoidCallback onDismiss;
+
+  @override
+  ConsumerState<MentionList> createState() => _MentionHandlerState();
+}
+
+class _MentionHandlerState extends ConsumerState<MentionList> {
+  final _focusNode = FocusNode();
+  final _scrollController = ScrollController();
+
+  int _selectedIndex = 0;
+
+  @override
+  void initState() {
+    super.initState();
+    WidgetsBinding.instance.addPostFrameCallback((_) {
+      _focusNode.requestFocus();
+    });
+  }
+
+  @override
+  void dispose() {
+    widget.onDismiss();
+    _scrollController.dispose();
+    _focusNode.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    // All suggestions list
+    final suggestions = ref
+        .watch(mentionSuggestionsProvider((widget.roomId, widget.mentionType)));
+    if (suggestions == null) {
+      return ErrorWidget(L10n.of(context).loadingFailed);
+    }
+    final menuWidget = Column(
+      mainAxisSize: MainAxisSize.min,
+      children: [
+        _buildMenuHeader(),
+        const Divider(height: 1, endIndent: 5, indent: 5),
+        const SizedBox(height: 8),
+        _buildMenuList(suggestions),
+      ],
+    );
+
+    return KeyboardListener(
+      focusNode: _focusNode,
+      onKeyEvent: (event) => _handleKeyEvent(event, suggestions),
+      child: menuWidget,
+    );
+  }
+
+  Widget _buildMenuHeader() => Padding(
+        padding: const EdgeInsets.all(8.0),
+        child: Text(
+          widget.mentionType == MentionType.user
+              ? L10n.of(context).users
+              : L10n.of(context).chats,
+        ),
+      );
+
+  Widget _buildMenuList(Map<String, String> suggestions) {
+    final String notFound = widget.mentionType == MentionType.user
+        ? L10n.of(context).noUserFoundTitle
+        : L10n.of(context).noChatsFound;
+
+    return Flexible(
+      child: suggestions.isEmpty
+          ? Padding(
+              padding: const EdgeInsets.all(16.0),
+              child: Text(notFound),
+            )
+          : ListView.builder(
+              shrinkWrap: true,
+              controller: _scrollController,
+              itemCount: suggestions.length,
+              itemBuilder: (context, index) {
+                final mentionId = suggestions.keys.elementAt(index);
+                final displayName = suggestions.values.elementAt(index);
+                final avatar = widget.mentionType == MentionType.user
+                    ? ref.watch(
+                        memberAvatarInfoProvider(
+                          (roomId: widget.roomId, userId: mentionId),
+                        ),
+                      )
+                    : ref.watch(roomAvatarInfoProvider(mentionId));
+                final options = widget.mentionType == MentionType.user
+                    ? AvatarOptions.DM(avatar, size: 18)
+                    : AvatarOptions(avatar, size: 28);
+
```

Comment:
inline with the switch-removing from above I'd make argue that the generic mention list should have an `AvatarOptions Function(String matchId, WidgetRef ref) avatarBuilder` given as a parameter ... this is pretty easy to provide from the outer thing as well...

Should still be using its own Consumer here though